opentargets_pharmgkb/evidence_generation.py

```diff
@@ -100,26 +156,46 @@ def chebi_id_to_iri(id_):
     return f'http://purl.obolibrary.org/obo/CHEBI_{id_}'
 
 
+def explode_and_map_phenotypes(df):
+    """
+    Maps phenotype text to EFO IRIs using Zooma. Explodes multiple phenotypes in single row.
+
+    :param df: dataframe to annotate (should have a 'Phenotype(s)' column)
+    :return: dataframe with 'efo' column added
+    """
+    df['Phenotype(s)'].fillna('', inplace=True)
+    split_phenotypes = explode_column(df, 'Phenotype(s)', 'split_phenotype')
+    with multiprocessing.Pool(processes=24) as pool:
+        str_to_iri = {
+            s: pool.apply(get_efo_iri, args=(s,))
+            for s in split_phenotypes['split_phenotype'].drop_duplicates().tolist()
+        }
+    mapped_phenotypes = pd.concat(
+        split_phenotypes[split_phenotypes['split_phenotype'] == s].assign(efo=none_to_nan(iri))
+        for s, iri in str_to_iri.items()
+    )
+    return mapped_phenotypes
+
+
 def generate_clinical_annotation_evidence(created_date, row):
     """Generates an evidence string for a PharmGKB clinical annotation."""
-    vcf_full_coords = get_coordinates_for_clinical_annotation(row['Variant/Haplotypes'], row['all_genotypes'])
     evidence_string = {
         # DATA SOURCE ATTRIBUTES
         'datasourceId': 'pharmgkb',
         'datasourceVersion': created_date,
 
         # RECORD ATTRIBUTES
         'datatypeId': 'clinical_annotation',
-        'studyId': row['Clinical Annotation ID'],
+        'studyId': row[ID_COL_NAME],
         'evidenceLevel': row['Level of Evidence'],
+        'literature': [str(x) for x in row['publications']],
 
         # VARIANT ATTRIBUTES
-        'variantId': vcf_full_coords,
+        'variantId': row['vcf_coords'],
         'variantRsId': row['Variant/Haplotypes'],
         'targetFromSourceId': row['ensembl_gene_id'],
```

Comment:
Correct me if I'm wrong, but doesn't this come from ``explode_and_map_genes()``, which gets information from the ``consequences_table``, and this one is generated from ``get_functional_consequences()``, which uses VEP? 

If so, I don't understand well the following bit:

> Note ``targetFromSourceId`` is mapped from the gene provided by PharmGKB, whereas ``variantOverlappingGeneId`` is what we get from VEP based on the variant definition.

Reply:
I think this is okay, `ensembl_gene_id` is indeed created in `explode_and_map_genes` which uses the `Gene` column (which is the HGNC symbol from PGKB directly).  `get_functional_consequences` uses `vcf_coords` to create the column `overlapping_gene`.  Maybe I'll change the name of the `ensembl_gene_id` column so it's less confusing...

Reply:
> which is the HGNC symbol from PGKB directly
This is the bit where I was confused, I probably misread the code. If that's the case, no need to change names, it was just me lost in the provenance of the gene ID.